sklearn/lda.py

```diff
@@ -450,13 +450,13 @@ def transform(self, X):
         X_new : array, shape (n_samples, n_components)
             Transformed data.
         """
-        check_is_fitted(self, ['xbar_', 'scalings_'], all_or_any=any)
-
-        X = check_array(X)
         if self.solver == 'lsqr':
             raise NotImplementedError("transform not implemented for 'lsqr' "
                                       "solver (use 'svd' or 'eigen').")
-        elif self.solver == 'svd':
+        check_is_fitted(self, ['xbar_', 'scalings_'], all_or_any=any)
```

Comment:
And further down `X_new = np.dot(X, self.scalings_)` makes me conclude that at least `scalings_` is indispensable?